test/testcafe/spec/DeviceChallengePollViewFailure_spec.js

```diff
@@ -34,32 +35,73 @@ const mock = RequestMock()
     'access-control-allow-methods': 'POST, OPTIONS'
   });
 
-fixture('Device Challenge Polling View with Polling Failure')
-  .requestHooks(logger, mock);
+const deviceInvalidatedErrorMsg = RequestMock()
+  .onRequestTo(/\/idp\/idx\/introspect/)
+  .respond(identifyWithDeviceProbingLoopback)
+  .onRequestTo(/\/idp\/idx\/authenticators\/poll/)
+  .respond((req, res) => {
+    return new Promise((resolve) => setTimeout(function() {
+      res.statusCode = '400';
+      res.setBody(errorDeviceInvalid);
+      resolve(res);
+    }, Constants.TESTCAFE_DEFAULT_AJAX_WAIT + 2000));
```

Comment:
I'm not that familiar with how polling works, but do we need this mandatory two second (plus `TESTCAFE_DEFAULT_AJAX_WAIT` **3** second) delay? Via mocking, I'd hope we could avoid unnecessary wait times.

@jhe-okta or @pradeepdewda-okta - Any ideas/suggestions?

Reply:
I think he is trying to mock actual `polling`.. There is no effective way in testcafe to mock `setTimeout` or `setInterval` as far as i know.

Reply:
We need some time to do polling here

Reply:
Can we create a ticket to investigate this?

Ideally, we should avoid any test execution delays, like this **5 second wait**. I'd hope after more investigation we can solve this via mocking and/or ensuring our tests are running in parallel so the burden isn't as high for our developers.

Reply:
There is one i filled a while ago.. 

https://oktainc.atlassian.net/browse/OKTA-430216

Reply:
one more similar ticket - https://oktainc.atlassian.net/browse/OKTA-365670